web/src/user_card_popover.js

```diff
@@ -55,6 +55,57 @@ let current_user_sidebar_popover;
 let current_user_sidebar_user_id;
 let userlist_placement = "right";
 
+// Shared prototype for popover instances
+const popover_prototype = {
+    instance: null,
+    is_open() {
+        return Boolean(this.instance);
+    },
+    hide() {
+        if (this.is_open()) {
+            this.instance.destroy();
+            this.instance = undefined;
+        }
+    },
+    handle_keyboard(key) {
+        if (!this.is_open()) {
+            blueslip.error("Trying to get the items when popover is closed.");
+            return;
+        }
+
+        const $popover = $(this.instance?.popper);
+        if (!$popover) {
+            blueslip.error("Cannot find popover data.");
+            return;
+        }
+
+        const $items = $("li:not(.divider):visible a", $popover);
+        popover_items_handle_keyboard(key, $items);
+    },
+};
+
+// This should be the main API for this module to work with popovers.
+export const popovers = {
+    user_group: {
+        ...popover_prototype,
+    },
+    manage_menu: {
+        ...popover_prototype,
+    },
+};
+
+export function any_active() {
+    return Object.values(popovers).some((instance) => instance.is_open());
+}
+
+export function hide_all_instances() {
+    for (const key in popovers) {
+        if (popovers[key].hide) {
+            popovers[key].hide();
+        }
+    }
+}
```

Comment:
This commit should probably be moved later and squashed with whatever uses it the first time; it's a lot easier to review the diff when one can see the before/after code that is migrated to it.